### DISCUSSION TITLE
Can't use with React-Native-Web/Expo Web

### DISCUSSION THREAD

**Discussion by arthur-fontaine** [pmndrs/zustand#2916] [Bug report]
The following line uses `import.meta` which seems to be incompatible with React Native Web / Expo Web (see https://github.com/expo/expo/issues/30323).

https://github.com/pmndrs/zustand/blob/51c00d22c34de84e6d17c53c32fd142a6f5d6543/src/middleware/devtools.ts#L164

It completely crashes my website (nothing shows).

In the mentioned issues, some have exported a specific umd version of their packages to address this issue. I don't know if it is something that you can consider.

**Comment by dai-shi** [comment]
Related reports:
- https://github.com/pmndrs/zustand/discussions/1967
- https://github.com/pmndrs/jotai/discussions/1717

**Comment by dai-shi** [comment]
Can you somehow use CJS build? That's where we remove `import.meta`.

**Reply by jonathanj** [reply]
The issue arises when needing to enable package.json `exports` support, which means that the resolving field is no longer `main`, and thus no longer CJS.

**Reply by jonathanj** [reply]
Seems like [this](https://github.com/pmndrs/zustand/discussions/1967#discussioncomment-9578159) is probably the only option, unfortunately.

**Reply by dai-shi** [reply]
We could add "react-native" condition pointing to CJS, but I'm not very confident if it's the right decision as it would prevent RN users to use ESM in the future.

**Reply by jonathanj** [reply]
Might also cause more harm than good, because not only would you need to navigate your way around this problem in general, but you would somehow need to undo the fix for some libraries that go out of their way to workaround it.

**Comment by gianpaj** [comment]
I upgraded from version 4.5.5 to 5.0.2 and it worked for me!

I have
```
expo 52.0.19
macOS Sonoma 14.7
```

**Comment by bryanprimus** [comment]
for some reason this is still happening for me

"zustand": "^5.0.5"

<img width="637" alt="image" src="https://github.com/user-attachments/assets/a246c1f9-7bff-4a2c-a006-90b97f11642e" />

**Reply by bryanprimus** [reply]
nvm i solved it using https://github.com/pmndrs/zustand/discussions/1967#discussioncomment-12801707
